python_ta/typecheck/base.py

```diff
@@ -145,6 +145,23 @@ def __str__(self):
         return string
 
 
+class TypeFailFunction(TypeFail):
+    """
+    TypeFailFunction occurs when a function is called with different arguments than expected.
+    Should be initialized with the tnode of the expected function type signature, and astroid
+    node where invalid function call occurred
+    """
+    def __init__(self, func_types: Tuple[CallableMeta], funcdef_node: astroid.FunctionDef, src_node: NodeNG, arg_index: List[int] = []):
+        self.func_types = func_types
```

Comment:
It's not clear what these instance attributes are. You should document them carefully. Also `arg_index` should be pluralized.

Finally, don't use mutable objects as default arguments.